lib/hammer_cli_foreman_puppet/commands.rb

```diff
@@ -0,0 +1,55 @@
+module HammerCLIForemanPuppet
+  RESOURCE_NAME_MAPPING = {
+    :puppetclass => :puppet_class,
+    :puppetclasses => :puppet_classes,
+    :environment => :puppet_environment,
+    :environments => :puppet_environments
+  }.freeze
+
+  RESOURCE_ALIAS_NAME_MAPPING = {
```

Comment:
Also, since we need to use this, you'll also need similar concern [1].

and include [2] it where `puppet-environment` options should be present.

[1] - https://github.com/Katello/hammer-cli-katello/blob/master/lib/hammer_cli_katello/lifecycle_environment_name_mapping.rb
[2] - https://github.com/Katello/hammer-cli-katello/blob/cba821db5b85c2e6d8e7085bcc6e6a5d93b3827b/lib/hammer_cli_katello/activation_key.rb#L9

Reply:
Thanks, I've added a similar concern. 
Do you have any more candidates for including it (other than `environment` itself)?

Reply:
Well, according to the `hammer full-help` command, the following commands have `--puppet-environment` option:

 - hammer > capsule > import-classes
 - hammer > discovery > provision (that's one from the hammer-cli-foreman-discovery plugin; not sure what we should do with this yet since it depends on hammer-cli-foreman)
 - hammer > host > create
 - hammer > host > list
 - hammer > host > update
 - hammer > hostgroup > create
 - hammer > hostgroup > update
 - hammer > location > add-environment
 - hammer > location > create
 - hammer > location > remove-environment
 - hammer > location > update
 - hammer > organization > add-environment
 - hammer > organization > create
 - hammer > organization > remove-environment
 - hammer > organization > update
 - hammer > policy > hosts
 - hammer > proxy > import-classes
 - hammer > puppet-class > info
 - hammer > puppet-class > list
 - hammer > puppet-environment > sc-params
 - hammer > sc-param > list
 - hammer > template > combination > create
 - hammer > template > combination > info
 - hammer > template > combination > update
 
 Hmm... I thought there will be quite a huge list... Maybe it's also worth to check with `--environment` option. Maybe there are options that should be even renamed.

Reply:
This `full-help` option is really nice, I have never noticed it before

Reply:
@amirfefer If you run the `hammer full-help` command for the hammer-cli-foreman master branch and also for your puppet extraction branch with the puppet plugin, there is an amazing output of all the commands that still need fixing. I think it is very handy!

Reply:
There are still some issues here. IMHO, we could still merge the PR, then we had a solid base and do bug fixing from here. 
@ofedoren What do you think?

Reply:
Probably we could.

FYI, I'm currenly working on https://github.com/theforeman/hammer-cli/pull/349 which might help with option building here and might fix the issues, didn't test though... I have a patch for Katello (https://github.com/Katello/hammer-cli-katello/pull/811) and maybe something similar will help here.

Reply:
Wow, looks like a huge change. Maybe we could have one of the `Q&A` meetings eventually, where you explain it in more detail?